pkgs/applications/window-managers/i3/default.nix

```diff
@@ -29,27 +29,41 @@ stdenv.mkDerivation rec {
     libstartup_notification libX11 pcre2 libev yajl xcb-util-cursor perl pango
     perlPackages.AnyEventI3 perlPackages.X11XCB perlPackages.IPCRun
     perlPackages.ExtUtilsPkgConfig perlPackages.InlineC
-    xorgserver xvfb-run
+    xorgserver xvfb-run xdotool xorg.setxkbmap xorg.xrandr which
   ];
 
   configureFlags = [ "--disable-builddir" ];
 
   postPatch = ''
     patchShebangs .
+
+    # This testcase generates a Perl executable file with a shebang, and
+    # patchShebangs can't replace a shebang in the middle of a file.
+    if [ -f testcases/t/318-i3-dmenu-desktop.t ]; then
```

Comment:
Do we need to check if the test case exists?

Reply:
Then I would need to add
```nix
  postPatch = ''
    patchShebangs .
  '';
```
to `i3/rounded.nix` (overwriting `oldAttrs.postPatch` instead of appending as usual) to make `i3-rounded` build again.  Would that be better?